# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1529954452 14400
#      Mon Jun 25 15:20:52 2018 -0400
# Node ID 01eb7f87eecd0803d2fa642fccb878bb5ac7ef81
# Parent  92a64cb5e4753945476a517a44f3bbe8547236e6
provide wrapper for mkostemps

* mkostemps-wrapper.h, mkostemps-wrapper.c: New files.
* liboctave/wrappers/module.mk: Update.

diff --git a/liboctave/wrappers/mkostemps-wrapper.c b/liboctave/wrappers/mkostemps-wrapper.c
new file mode 100644
--- /dev/null
+++ b/liboctave/wrappers/mkostemps-wrapper.c
@@ -0,0 +1,40 @@
+/*
+
+Copyright (C) 2016-2018 John W. Eaton
+
+This file is part of Octave.
+
+Octave is free software: you can redistribute it and/or modify it
+under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+Octave is distributed in the hope that it will be useful, but
+WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with Octave; see the file COPYING.  If not, see
+<https://www.gnu.org/licenses/>.
+
+*/
+
+// mkostemp may be provided by gnulib.  We don't include gnulib headers
+// directly in Octave's C++ source files to avoid problems that may be
+// caused by the way that gnulib overrides standard library functions.
+
+#if defined (HAVE_CONFIG_H)
+#  include "config.h"
+#endif
+
+#include <fcntl.h>
+#include <stdlib.h>
+
+#include "mkostemps-wrapper.h"
+
+int
+octave_mkostemps_wrapper (char *tmpl, int suffixlen)
+{
+  return mkostemps (tmpl, suffixlen, O_BINARY);
+}
diff --git a/liboctave/wrappers/mkostemps-wrapper.h b/liboctave/wrappers/mkostemps-wrapper.h
new file mode 100644
--- /dev/null
+++ b/liboctave/wrappers/mkostemps-wrapper.h
@@ -0,0 +1,36 @@
+/*
+
+Copyright (C) 2016-2018 John W. Eaton
+
+This file is part of Octave.
+
+Octave is free software: you can redistribute it and/or modify it
+under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+Octave is distributed in the hope that it will be useful, but
+WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with Octave; see the file COPYING.  If not, see
+<https://www.gnu.org/licenses/>.
+
+*/
+
+#if ! defined (octave_mkostemps_wrapper_h)
+#define octave_mkostemps_wrapper_h 1
+
+#if defined __cplusplus
+extern "C" {
+#endif
+
+extern int octave_mkostemps_wrapper (char *tmpl, int suffixlen);
+
+#if defined __cplusplus
+}
+#endif
+
+#endif
diff --git a/liboctave/wrappers/module.mk b/liboctave/wrappers/module.mk
--- a/liboctave/wrappers/module.mk
+++ b/liboctave/wrappers/module.mk
@@ -10,16 +10,17 @@ NOINSTALL_WRAPPERS_INC = \
   %reldir%/gen-tempname-wrapper.h \
   %reldir%/getopt-wrapper.h \
   %reldir%/glob-wrappers.h \
   %reldir%/hash-wrappers.h \
   %reldir%/iconv-wrappers.h \
   %reldir%/localcharset-wrapper.h \
   %reldir%/math-wrappers.h \
   %reldir%/mkostemp-wrapper.h \
+  %reldir%/mkostemps-wrapper.h \
   %reldir%/nanosleep-wrapper.h \
   %reldir%/nproc-wrapper.h \
   %reldir%/octave-popen2.h \
   %reldir%/putenv-wrapper.h \
   %reldir%/set-program-name-wrapper.h \
   %reldir%/signal-wrappers.h \
   %reldir%/stat-wrappers.h \
   %reldir%/strdup-wrapper.h \
@@ -51,16 +52,17 @@ WRAPPERS_SRC = \
   %reldir%/gen-tempname-wrapper.c \
   %reldir%/getopt-wrapper.c \
   %reldir%/glob-wrappers.c \
   %reldir%/hash-wrappers.c \
   %reldir%/iconv-wrappers.c \
   %reldir%/localcharset-wrapper.c \
   %reldir%/math-wrappers.c \
   %reldir%/mkostemp-wrapper.c \
+  %reldir%/mkostemps-wrapper.c \
   %reldir%/nanosleep-wrapper.c \
   %reldir%/nproc-wrapper.c \
   %reldir%/octave-popen2.c \
   %reldir%/putenv-wrapper.c \
   %reldir%/set-program-name-wrapper.c \
   %reldir%/signal-wrappers.c \
   %reldir%/stat-wrappers.c \
   %reldir%/strdup-wrapper.c \
